service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentService.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.enums.PartyType;
+import uk.gov.hmcts.reform.fpl.model.RespondentParty;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class RespondentService {
+
+    @Autowired
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    public AboutToStartOrSubmitCallbackResponse setMigratedValue(CaseDetails caseDetails) {
+        Map<String, Object> data = caseDetails.getData();
+
+        if (caseDetails.getData().containsKey("respondents1") || !caseDetails.getData().containsKey("respondents")) {
+            data.put("respondentsMigrated", "Yes");
+
+            if (!caseDetails.getData().containsKey("respondents1")) {
+                // Populates first respondent so UI contains expanded Respondent Object.
+                List<Map<String, Object>> populatedRespondent = new ArrayList<>();
+                populatedRespondent.add(ImmutableMap.of(
+                    "id", UUID.randomUUID().toString(),
+                    "value", ImmutableMap.of(
+                        "party", ImmutableMap.of(
+                            // Variable within CCD party structure must be set to expand Collection.
+                            // PartyId is a hidden field so setting a value will not persist to the db
+                            "partyId", UUID.randomUUID().toString()
```

Comment:
Should this not be partyID rather than partyId?

Reply:
Good spot but does not really matter as value is never used / persisted.